### ISSUE TITLE
setInterval vs setTimeout

### ISSUE THREAD

**Issue by dugwood** [devbridge/jQuery-Autocomplete#575]
Just asking myself why you make use of `setInterval` instead of `setTimeout`? As far as I can see, your interval is always cleared, so there's no need for repetition, and `setTimeout` may be safer.

PS: I may have to add a `deferRequestBy` parameter that would defer the next request. For example, instead of waiting for a second key press or `deferRequestBy`'s timeout, it will show the results instantly for the first letter, then allow the second request to be sent only after a short time, even if another key is pressed. Should I add another parameter, or can I use the `deferRequestBy`, for example with a negative value?

**Comment by tkirda** [comment]
It can be safely changed to `setTimeout`, no reason for `setInterval`.

Regarding deferring the request, it would be micro optimization, if user types quickly results still would be discarded. And if using local lookup, it does not make sense to have it at all.

**Comment by dugwood** [comment]
Thanks @tkirda.

For the deferring, my goal was to limit requests to the server (say 4/sec, so a timeout of 250ms), but if the user types too quickly, there's no answer as long as he types down under the timeout. And it seems that users have been used to seeing things displaying... even if they don't have finished typing.
So maybe `deferRequestBy` should be changed with that in mind: first request is sent immediately, but next requests are deferred. So the user sees that there's something happening.

For now I'll reduce my current setting from 600ms to 250ms, that should do the trick for my users (600ms is a bit too much).
